### ISSUE TITLE
Any plans to make a MacBook M1 (ARM) version available?

### ISSUE THREAD

**Issue by sofakng** [litchie/dospad#104]
Do you have any plans to compile a version of this for the new MacBook M1 (ARM) chips?

It seems like this might be the perfect project to have a version of DosBox running native on the M1 ARM CPUs?

**Comment by litchie** [comment]
But you can just compile the official dosbox version. There is no need for the virtual keyboard/gamepad support as iDOS does.

**Comment by DominusExult** [comment]
I'm doing builds from the official SVN of DOSBox and these can be downloaded from https://www.dropbox.com/s/kbrf5elbcu66kst/Dosbox-Snapshot.dmg?dl=0. These are code signed and notarized.
The Rosetta2 emulation on the Apple Silicon is outstanding and these builds run quite nicely.
A native version is worked on but still runs into some problems.

**Comment by sofakng** [comment]
@DominusExult Thanks for the information ... I understand about Rosetta 2 emulation (and it does work amazing) but I think it would be nice to have an ARM (M1) version to further improve performance (and battery).

Do you have a Git I can follow?  I'm wondering what issues you are having with the native version.  Last night I started compiling some dependencies (SDL2) and that went well and I wasn't aware of any other DosBox dependencies.

**Comment by DominusExult** [comment]
With an M1 you will have more than enough performance. Other than iDos, you will have a fully working dynamic core which is really really fast already.

I'm using main Dosbox which uses SDL 1.2x but the dependencies are not a problem.
The binary runs fine, only as soon as you put it in an app bundle it crashes as soon as you switch to dynamic core (mostly automatically once a demanding game starts or by entering "core dynamic" on the prompt). Some kind of security takes hold and so far I couldn't fix this on the DTK.
If you have an M1 devixzyou can give it a try https://www.dropbox.com/s/hy3vbsr0sv5aoyl/DOSBox-snapshotARM64.dmg?dl=0
Try entering "core dynamic" and please report back whether it crashes or not. I would really like to know if that is a quirk of the DTK vs a real M1 machine.

**Comment by sofakng** [comment]
It crashed:

Exception Type:        EXC_BAD_ACCESS (SIGBUS)
Exception Codes:       KERN_PROTECTION_FAILURE at 0x0000000132ee8000
Exception Note:        EXC_CORPSE_NOTIFY

Did you try to disable SIP (System Integrity Protection) and see if that's
causing the problem?

**Comment by DominusExult** [comment]
Thanks for testing, now that you tested it, it must be something in the patch for ARM that triggers this.
If it were the SIP the binary wouldn't run at all, I guess.
You can give this a try if you want https://www.dropbox.com/s/df9lptx6y4echdj/dosbox_arm64.zip?dl=0

**Comment by sofakng** [comment]
Same here and in the console window is also shows this:

zsh: bus error  /Users/sofakng/Downloads/dosbox_arm64

**Comment by DominusExult** [comment]
Please try https://www.dropbox.com/s/hy3vbsr0sv5aoyl/DOSBox-snapshotARM64.dmg?dl=0 again, it's a new version that should work *ONCE* and not until a reboot :)
It has a hack to circumvent the creation of the tempfile. The hack has it hardcoded to /tmp/DOSBox31233 and it persists until a reboot. You can of course delete it, too, so you don't have to reboot :)
But it's just a hack test, nothing to really use :)

**Comment by sofakng** [comment]
Thanks!  Yes, it worked exactly once like you described.

Does that you mean you know where the problem is located and can possibly
fix it?

Thanks for the work on this!  Also, feel free to e-mail me directly (
jklimek@gmail.com) so we can stop interrupting litchie's github :)

**Comment by DominusExult** [comment]
Good to know it works (with the big caveat).
Jmarsh is actually the one working on it, I'm just the guy with the machine to compile and test it. We are getting closer to the culprit but still hitting a wall ;)
And yes, I'll contact you directly from now on ;)

**Comment by MisutaaUrufu** [comment]
Don't M1 Macs have the ability to run iOS software natively? I don't have an M1 Mac to test, but wouldn't it be possible to just... build and run the iDOS/dospad project on an M1 Mac without further configuration?

**Comment by litchie** [comment]
Running iDOS on M1 macbook:
https://www.youtube.com/watch?v=Y5I3KbRkywQ
